### PR TITLE
Python with Pybind11

### PR DESCRIPTION
**Pull request by mrpollo**
all credit here goes to @julianoes I only cleaned up the Make files

## Build
```
make python
```

## Install (to site-packages)
```
make python install
```

## Tests
**NOTE:** Tests will only run if you have SITL running
```
make python_test
```

### CODE REVIEW

File: python_pybind_test.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/env python
```
Comment (hamishwillee) [inline python_pybind_test.py]: @mrpollo @julianoes This is functionally the same example as for C++. We're going to want it as an example, in a sensible folder structure that identifies the test programming language and the test name (allowing us to have multi-file tests). Ideally this should match the C++ test structure.

**Comment by julianoes** [comment]
Note to self: need to upload the new Docker image for the Python dependency.

File: python_pybind_test.py (hunk 1 above)
Comment (mrpollo) [reply python_pybind_test.py]: agreed, I added a commit to move it

**Comment by mrpollo** [comment]
Updated this PR, python directory will hold all python files from now on, we only have a test for now, this is no longer a WIP.

**NOTE** if we decide to move on with gRPC we should figure out what it means for pybind.

**Comment by mrpollo** [comment]
@julianoes we need to decide if we are going to be using pybind11 or not, I think it's a great feature to have, but if we are not going to be able to properly maintain it we shouldn't add it, I also think gRPC would deprecate this, whats your take?

**Comment by julianoes** [comment]
@mrpollo you are right. It looks like gRPC will take care of this.

**Pull request closed without merging**